packages/bot-strategy-runner/src/index.ts

```diff
@@ -0,0 +1,213 @@
+import program from "commander";
+import nodeFetch from "node-fetch";
+import assert from "assert";
+import * as fs from "fs";
+import * as cliProgress from "cli-progress";
+import * as bluebird from "bluebird";
+
+import { Logger, delay } from "@uma/financial-templates-lib";
+
+import { strategyRunnerConfig, buildBotConfigs, buildGlobalWhitelist, mergeConfig } from "./ConfigBuilder";
+import { runBot } from "./BotEntryWrapper";
+
+// Defaults for bot execution. If the user does not define these then these settings will be used.
+const defaultPollingDelay = 300;
+const defaultStrategyTimeout = 120;
+const defaultBotConcurrency = 10;
+const defaultNetwork = "mainnet_mnemonic";
+
+// Global progress bar to show the status of the strategy runner during it's execution.
+let counter = 0;
+const progressBar = new cliProgress.SingleBar(
+  { format: "[{bar}] {percentage}% | bots executed: {value}/{total} | recent execution: {botIdentifier}" },
+  cliProgress.Presets.shades_classic
+);
+
+// Main entry point that takes in a strategyRunnerConfig and executes bots within an execution while loop.
+async function runStrategies(strategyRunnerConfig: strategyRunnerConfig) {
+  strategyRunnerConfig = _setConfigDefaults(strategyRunnerConfig); // Default important configs. User configs take preference.
+  Logger.debug({
+    at: "BotStrategyRunner",
+    message: "Starting bot strategy runner",
+    strategyRunnerConfig: strategyRunnerConfig
+  });
+
+  // Generate a global whitelist of addresses that all enabled bots will run on.
+  const globalWhiteList = await buildGlobalWhitelist(strategyRunnerConfig);
+
+  // Construct bot configs for all enabled bots. This includes all bot specific overrides, whitelists, blacklists and any
+  // extra settings defined in the config. See the readme for full details on what is possible with this config.
+  const allBotsConfigs = await buildBotConfigs(globalWhiteList, strategyRunnerConfig);
+  console.log("allBotsConfigs", allBotsConfigs);
+
+  Logger.debug({
+    at: "BotStrategyRunner",
+    message: "Constructed global bot settings and whitelist",
+    concurrency: strategyRunnerConfig.botConcurrency,
+    globalWhiteList,
+    totalBotsToExecute: allBotsConfigs.length
+  });
+
+  for (;;) {
+    Logger.debug({
+      at: "BotStrategyRunner",
+      message: "Executing set of bots concurrently",
+      concurrency: strategyRunnerConfig.botConcurrency
+    });
+
+    progressBar.start(allBotsConfigs.length, 0);
+
+    // Execute all bots in a bluebird map with limited concurrency. Note that none of the `runBot` calls ever throw errors.
+    // Rather, they indicate an error via an `error` key within the response. The promise.race between the `runBot` and
+    // _rejectAfterDelay bounds how long a given strategy can run for.
+    const executionResults = await (bluebird as any).map(
```

Comment:
Why do you have to do this cast? Do the bluebird types not contain the map function?

Reply:
otherwise I get this error:
![image](https://user-images.githubusercontent.com/12886084/114994912-94e4aa80-9e9d-11eb-9a2d-3415a4433241.png)


I _have_ installed `"@types/bluebird"` so I'm not sure why this is the case. will do some more playing

Reply:
ok I was able to address this by changing the import type. now imported as `import bluebird from "bluebird";` and it works as expected :)